libs/dkg/src/beacon_member.cpp

```diff
@@ -0,0 +1,160 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "dkg/beacon_manager.hpp"
+
+namespace fetch {
+namespace dkg {
+
+void BeaconManager::Reset(uint64_t cabinet_size, uint32_t threshold)
+{
+  cabinet_size_ = cabinet_size;
+  threshold_    = threshold;
+
+  received_shares_.clear();
+  participants_.clear();
+
+  verification_vectors_.resize(cabinet_size);
+  received_shares_.resize(cabinet_size);
+
+  // Creating member id
+  // TODO(tfr): Make loadable
+  certificate_.reset(new ECDSASigner());
+  certificate_->GenerateKeys();
+  auto id = crypto::bls::PrivateKeyByCSPRNG();
+  id_.v   = id.v;
```

Comment:
Is it possible to avoid single character variable names?

Reply:
Ah I see this is actually from the BLS library

Reply:
No, unfortunately I did not invent this.